src/Common/src/Interop/OSX/System.Native/Interop.ProcPidInfo.cs

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        // Constants from sys/param.h
+        private const int MAXCOMLEN = 16;
```

Comment:
Are we validating these values somewhere in the native shim?  Seems like otherwise this code still has an implicit dependency on the OS implementation.